### PR TITLE
Add Booster in battle

### PR DESCRIPTION
**Pull request by sonohoshi**
### Description

SSIA

### Features that work now

1. Boost button
2. Boost Popup
3. Use AP all at once and enter the stage

### How to test

1. Enter to Quest preparation UI at cleared stage.
2. Click Booster button.
3. Enter to stage using Booster.

### Related Links

[[전투준비] 전투&미미르 전투 시작 & 부스트 버튼 제작](https://app.asana.com/0/958521740385861/1200968125118328/f)
[[전투준비] 부스트 설정 팝업 제작](https://app.asana.com/0/958521740385861/1200968125118330/f)
and etc.

### Screenshot

image
image
image

### CODE REVIEW

**Comment by Namyujeong** [comment]
Good!

@sonohoshi Even if you have more AP, the maximum value of the AP that can be used at once should be 60. Therefore, the maximum value of the bar is 12 to sum up.
image

@BBeom06 The text on the result screen is weird. Please check it out.
스크린샷 2021-09-17 오전 12 00 36

**Comment by sonohoshi** [comment]
> Even if you have more AP, the maximum value of the AP that can be used at once should be 60. Therefore, the maximum value of the bar is 12 to sum up.

- OK. I checked it and will fix it.

> The text on the result screen is weird. Please check it out.

- This is the part I didn't fix during the test. I'll fix it.

**Comment by Namyujeong** [comment]
```
UI_BOOSTER_BUTTON,Booster,부스터,Booster.,,増幅器,Booster,Aumentador de presión,ผู้สนับสนุน,Wzmacniacz,Booster.,Booster,Pemacu,Booster,Бустер,ការរបបើកចិត្តអាឃាដ,,助推器,
UI_BOOSTER_POPUP_TITLE,Battle Booster,전투 부스터,Battle Booster.,,バトルブースター,Battle Booster.,Battle Booster,Battle Booster,Booster bitwy,Battle Booster.,Battle Booster,Battle Booster.,Booster,Боевой усилитель,bowom boster ប្រយុទ្ធ,,战斗助推器,
UI_BOOSTER_POPUP_DESCRIPTION,"You can consume AP at once,
and receive rewards and exp at once.","행동력을 한꺼번에 사용하여 
리워드와 경험치를 한꺼번에 받을 수 있습니다.","Você pode consumir AP de uma vez
e receber recompensas e exp de uma só vez.",,"あなたは一度にAPを消費することができます、
そして一度に報酬とexpを受け取ります。","Sie können ap auf einmal verbrauchen,
und empfangen Sie Belohnungen und EXP sofort.","Puedes consumir AP a la vez,
y recibir recompensas y expiértase a la vez.","คุณสามารถกิน AP ได้ทันที
และรับรางวัลและประสบการณ์ในครั้งเดียว","Możesz spożywać ap jednocześnie,
i otrzymywać nagrody i exp.","Galite vartoti AP vienu metu,
ir gauti atlygį ir per vieną kartą.","Je kunt ap ineen consumeren,
en ontvang beloningen en EXP in één keer.","Anda dapat mengkonsumsi AP sekaligus,
dan terima hadiah dan exp sekaligus.","Vous pouvez consommer AP à la fois,
et recevoir des récompenses et des dépenses à la fois.","Вы можете потреблять AP одновременно,
и получать награды и опыт сразу.","អ្នកអាចទទួលបាន ap ap ក្នុងពេលតែមួយ,
ហើយទទួលបានរង្វាន់និង Exp ក្នុងពេលតែមួយ។",,"您可以立即使用AP，
并立即获得奖励和exp。",
UI_BOOSTER_OWN_AP,Own,보유,Ter,,自分の,Besitzen,Propio,เป็นเจ้าของ,Własny,Savo,Eigen,Memiliki,Posséder,Собственный,នៃកម្ផសិត្ធិ,,自己的,
UI_BOOSTER_NEED_AP,Need,필요,Necessidade,,必要,Brauchen,Necesitar,ความต้องการ,Potrzebować,Reikia,Nodig hebben,Membutuhkan,Avoir besoin,Необходимость,រតុវការ,,需要,
UI_BOOSTER_USING_AP,{0}AP,,,,,,,,,,,,,,,,,
```

**Comment by Namyujeong** [comment]
There is a bug with toggle activated even though I don't make a reservation to exit.
스크린샷 2021-09-27 오후 6 27 12

**Comment by Namyujeong** [comment]
character spine was disappeared when open battle boost popup
image

**Comment by Namyujeong** [comment]
Please fix it so that it is labeled 50AP instead of 10.
image

**Comment by Namyujeong** [comment]
- Just before the "Booster Complete" text appeared, I saw "Perfect" briefly.
- 재료가 독립시행으로 획득되고 있는게 맞나요? 한번 전투 * 10배로 재료가 떨어지고 있는 것 같네요. [기획 문서](https://www.notion.so/planetarium/d6d58782022347d1b1a3a7fdff1d4af5#e2e7e746e15b46ab8c196fa28b1f8973)를 참조 부탁드립니다. cc @hskim881028 
스크린샷 2021-09-27 오후 6 42 07

File: nekoyume/Assets/_Scripts/UI/Popup/BoosterPopup.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        {
+            _stage = stage;
+            _costumes = costumes;
+            _equipments = equipments;
+            _consumables = consumables;
+            _player = _stage.GetPlayer(Vector3.zero);
+
+            ReactiveAvatarState.ActionPoint.Subscribe(value =>
+            {
+                apSlider.maxValue = value / 5 >= 12 ? 12 : value / 5;
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Popup/BoosterPopup.cs:67]: 5 should refer to the action point on the stage sheet.
12 seems to be better to use by declaring it as const somewhere.

File: nekoyume/Assets/_Scripts/UI/Popup/BoosterPopup.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                needAPText.text = (Game.Game.instance
+                    .TableSheets.StageSheet.Values.FirstOrDefault(i =>
+                        i.Id == Find<WorldMap>().SelectedStageId).CostAP * value).ToString();
+            });
+
+            var actionPoint = Game.Game.instance.States.CurrentAvatarState.actionPoint;
+            ownAPText.text = actionPoint.ToString();
+            // Call onValueChanged by Change value
+            apSlider.value = 0;
+            apSlider.value = apSlider.maxValue = actionPoint / 5 >= 12 ? 12 : actionPoint / 5;
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Popup/BoosterPopup.cs:83]: 5 should refer to the action point on the stage sheet.
12 seems to be better to use by declaring it as const somewhere.

File: nekoyume/Assets/_Scripts/UI/BattleResult.cs [hunk 3]
```diff
@@ -700,6 +715,12 @@ private void StopVFX()
                 _battleWin03VFX = null;
             }
 
+            if (_battleWin04VFX)
+            {
+                _battleWin04VFX.Stop();
+                _battleWin04VFX.Stop();
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/BattleResult.cs:721]: duplicate code

File: nekoyume/Assets/_Scripts/Game/Stage.cs [hunk 4]
```diff
@@ -90,6 +90,7 @@ public class Stage : MonoBehaviour, IStage
         public bool IsExitReserved { get; set; }
         public bool IsRepeatStage { get; set; }
         public bool IsAvatarStateUpdatedAfterBattle { get; set; }
+        public int PlayCount { get; set; }
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/Game/Stage.cs:95]: spaces should be deleted

**Comment by Namyujeong** [comment]
AP부족시 출력할 한줄팝업 레이블

```
UI_NOT_ENOUGH_AP,Insufficient AP,AP가 부족합니다,AP insuficiente,,APが不足している,Unzureichende AP,Insuficiente AP,AP ไม่เพียงพอ,Niewystarczający AP,Nepakankamas AP,Onvoldoende AP,AP tidak mencukupi,AP insuffisant,Недостаточно AP,AP មិនគ្រប់គ្រាន់,,AP不足,
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 5]
```diff
@@ -700,8 +700,9 @@ private void ResponseHackAndSlash(ActionBase.ActionEvaluation<HackAndSlash> eval
                     Game.Game.instance.TableSheets.CostumeStatSheet,
                     StageSimulatorVersionV100025
                 );
-                simulator.Simulate();
+                simulator.Simulate(eval.Action.playCount);
                 var log = simulator.Log;
+                Game.Game.instance.Stage.PlayCount = eval.Action.playCount;
```
[the remark this answers is not in the document]
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:703]: Yes. `PlayCount` reset when call `ResponseHackAndSlash()`.
And PlayCount referenced in `CoSpawnPlayer()` and `CoStageEnd()`. These methods called after calling `ResponseHackAndSlash()`.

**Comment by Namyujeong** [comment]
@sonohoshi 레이블이 없을리가 없는데.. 하면서 추가해드렸더니 역시 있었네요. 행동력 없을 시 Boost 버튼을 눌렀을때 뜨는 레이블이 Start 버튼 눌렀을때 뜨는것과 같도록 수정 부탁드립니다. (이번에 새로 드린 레이블 삭제 부탁드립니다)

그 외 스타일 수정은 내일 저랑 콜로 미세조정을 좀 해봅시다.

**Comment by hskim881028** [comment]
Conflicting files

**Pull request merged**